restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/GeolocationEndpoint.java

```diff
@@ -0,0 +1,732 @@
+        if (APPLICATION_XML_TYPE == responseType) {
+            final RestCommResponse response = new RestCommResponse(new GeolocationList(geolocations));
+            return ok(xstream.toXML(response), APPLICATION_XML).build();
+        } else if (APPLICATION_JSON_TYPE == responseType) {
+            return ok(gson.toJson(geolocations), APPLICATION_JSON).build();
+        } else {
+            return null;
+        }
+    }
+
+    protected Response deleteGeolocation(final String accountSid, final String sid) {
+        Account account;
+        try {
+            secure(account = accountsDao.getAccount(accountSid), "RestComm:Delete:Geolocation");
+            Geolocation geolocation = dao.getGeolocation(new Sid(sid));
+            if (geolocation != null) {
+                secure(account, geolocation.getAccountSid(), SecuredType.SECURED_APP);
+            }
+        } catch (final Exception exception) {
+            return status(UNAUTHORIZED).build();
+        }
+        dao.removeGeolocation(new Sid(sid));
+        return ok().build();
+    }
+
+    public Response putGeolocation(final String accountSid, final MultivaluedMap<String, String> data,
+                                   GeolocationType geolocationType, final MediaType responseType) {
+        Account account;
+        try {
+            account = accountsDao.getAccount(accountSid);
+            secure(account, "RestComm:Create:Geolocation", SecuredType.SECURED_APP);
+        } catch (final Exception exception) {
+            return status(UNAUTHORIZED).build();
+        }
+
+        try {
+            validate(data, geolocationType);
+        } catch (final NullPointerException nullPointerException) {
+            // API compliance check regarding missing mandatory parameters
+            return status(BAD_REQUEST).entity(nullPointerException.getMessage()).build();
+        } catch (final IllegalArgumentException illegalArgumentException) {
+            // API compliance check regarding malformed parameters
+            cause = illegalArgumentException.getMessage();
+            rStatus = responseStatus.Failed.toString();
+        } catch (final UnsupportedOperationException unsupportedOperationException) {
+            // API compliance check regarding parameters not allowed for Immediate type of Geolocation
+            return status(BAD_REQUEST).entity(unsupportedOperationException.getMessage()).build();
+        }
+
+        /*********************************************/
+        /*** Query GMLC for Location Data, stage 1 ***/
+        /*********************************************/
+        try {
+            String targetMSISDN = data.getFirst("DeviceIdentifier");
+            Configuration gmlcConf = configuration.subset("gmlc");
+            String gmlcURI = gmlcConf.getString("gmlc-uri");
+            URL url = new URL("http://"+gmlcURI+"/restcomm/gmlc/rest?msisdn="+targetMSISDN);
```

Comment:
Fixed as for 563ebaa (tested with success together with GMLC and generated new binaries)